src/sink/kafka.rs

```diff
@@ -149,16 +151,21 @@ impl KafkaMessageSender for FutureProducer<STFUContext> {
         topic: &str,
         payload: &[u8],
         key: &[u8],
+        headers: HashMap<Vec<u8>, Vec<u8>>,
         connection_id: Option<Uuid>
     ) -> BoxedKafkaPublishable {
+        let owned_headers = OwnedHeaders::new();
+        for (k, v) in headers {
+            owned_headers.clone().add(str::from_utf8(&k).unwrap(), &v);
+        };
         Box::new(KafkaPublishResult {
-            inner: Some(self.send_copy(
-                topic,
-                /* partition */ None,
```

Comment:
This seems lost in translation.  The new code will publish to a random partition?